### PR TITLE
Drop support for Kotlin/JS

### PR DESCRIPTION
**Pull request by def-gthill**
@neta-elad Requesting your review to make sure you're aware of this major change to the project structure. Please make sure you're working off this version if you make more changes to the CLI program, otherwise there will be tricky merge conflicts.

When I first set up the Lexurgy website, I used Kotlin's support for compiling to JavaScript to embed it directly into the website. But this feature was, and still is, experimental, and made maintaining the project increasingly difficult.

With this PR, I'm finally doing away with Kotlin/JS. All future versions of the website will process sound changes by make API calls to a server running Lexurgy on the JVM.

### CODE REVIEW

**Comment by neta-elad** [comment]
I've started looking around. I see that this is mostly deleting the `jsMain` and `jsTest` directories and moving `jvmMain` to `main`, but there also some small changes here and there that I'm trying to grasp. Do you also have a PR for the lexurgy-web repo? Or a new repo that implements the backend for lexurgy-web?

**Comment by neta-elad** [comment]
OK, so I have a few comments here and there but I think this mostly looks great. 

I think unifying the repos is the way to go, but I guess you'd want to do that separately from this PR. 

The main advantage of using Kweb would be not needing a separate frontend codebase in JS. It would also handle all of the API calls/networking communication automatically (and might be more efficient, since it uses Websockets).

**Comment by def-gthill** [comment]
> OK, so I have a few comments here and there but I think this mostly looks great.

And those comments are...?

> I think unifying the repos is the way to go, but I guess you'd want to do that separately from this PR.

Yeah, that'll be a separate PR.

> The main advantage of using Kweb...

I've already written a substantial amount in React, so I'm highly reluctant to switch everything over to a different framework, especially one that I just heard about.

**Comment by neta-elad** [comment]
Yes, makes sense.

I gave my comments in the code itself, doesn't it show them to you?

**Comment by neta-elad** [comment]
Sorry, forgot to submit my comments up until now!

File: src/main/kotlin/com/meamoria/lexurgy/MainJvm.kt
[the remark this answers is not in the document]
Comment (def-gthill) [reply src/main/kotlin/com/meamoria/lexurgy/MainJvm.kt:1]: Fixed

File: src/main/kotlin/com/meamoria/lexurgy/sc/SoundChanger.kt [hunk 1]
```diff
@@ -16,6 +17,26 @@ class SoundChanger(
         }
     }
 
+    /**
+     * The list of rule names that will be used in any tracing
+     * output, in the order that the rules get applied.
+     */
+    val ruleNames: List<String>
```
[the remark this answers is not in the document]
Comment (def-gthill) [reply src/main/kotlin/com/meamoria/lexurgy/sc/SoundChanger.kt]: Added

File: src/main/kotlin/com/meamoria/lexurgy/sc/SoundChanger.kt [hunk 2]
```diff
@@ -37,13 +58,15 @@ class SoundChanger(
         debugWords: List<String> = emptyList(),
         romanize: Boolean = true,
         debug: (String) -> Unit = ::println,
+        trace: (String, String, String) -> Unit = { _, _, _ -> },
```
[the remark this answers is not in the document]
Comment (def-gthill) [reply src/main/kotlin/com/meamoria/lexurgy/sc/SoundChanger.kt]: Switched to a parameter object here, with both pre- and post-transformation words. Further properties with default values could be added later.

**Pull request merged**